applications/mne_scan/plugins/fiffsimulator/fiffsimulatorproducer.cpp

```diff
@@ -146,7 +146,7 @@ void FiffSimulatorProducer::run()
     connectDataClient(m_pFiffSimulator->m_sFiffSimulatorIP);
 
     qint32 count = 0;
-    while(m_pRtDataClient->state() != QTcpSocket::ConnectedState) {
+    while(!isInterruptionRequested() && m_pRtDataClient->state() != QTcpSocket::ConnectedState) {
```

Comment:
While correct, operator precedence hacking is a bit too much hack-stylee. Would suggest use of perenthesis to make explicit what is implicit by the compiler rules.